File: guessing.java
import java.io.PrintWriter;
import java.util.Scanner;

/*
 * Guessing Game - guessing
 * Solution written by Joshua Linge
 */

/*
Sample input:
2
1 100 33
1 3 2

Sample output:
Game #1: 4 guesses
Game #2: 1 guess

 */
public class guessing {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		PrintWriter out = new PrintWriter(System.out);
		
		// Total number of games.
		int n = in.nextInt();
		
		// Loop through each game played.
		for(int curGame = 1; curGame <= n; ++curGame) {
			
			// Lower bound on Adam's chosen number range.
			int a = in.nextInt();
			
			// Upper bound on Adam's chosen number range.
			int b = in.nextInt();
			
			// Adam's chosen number.
			int x = in.nextInt();
			
			// Start our guessing range with the lower bound 
			// at a and our upper bound at b. 
			int lowerBound = a;
			int upperBound = b;
			
			// Start with zero guesses initially.
			int guessCount = 0;
			
			// Keep guessing until we find Adam's chosen number. 
			while(true) {
				// Take the average of our range, rounding up.
				int mid = (lowerBound + upperBound + 1) / 2;

				// Increase our guess count.
				++guessCount;
				
				if(mid == x) {			// Guess was correct, game is over.
					break;
				} else if(mid > x) {	// Guess was too high, decrease the upper bound.
					upperBound = mid-1;
				} else {				// Guess was too low, increase the lower bound.
					lowerBound = mid+1;
				}
			}
			
			// Print output.
			if(guessCount > 1) {
				out.println("Game #"+curGame +": "+guessCount +" guesses");
			} else { 
				out.println("Game #"+curGame +": 1 guess");
			}
		}
		
		out.close();
		in.close();
	}

}

File: guessing.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>
#include <cstdio>

using namespace std;

int main() {

	int n;
	cin >> n;

	//loop for each input case
	for(int game = 1; game <= n; game++){
		//input variables
		int a,b,x;
		cin >> a >> b >> x;

		int numberOfGuesses = 0;

		int min = a;
		int max = b;

		//simulating the guessing game
		while(true){
			numberOfGuesses++;
			//the plus one here ensures that the guess rounds up if min+max is odd
			int toguess = (min+max+1)/2;
			if(toguess == x){
				break;
			}
			else if(toguess > x){//the guess was too high so Jimmy knows the maximum number x could be is less than toguess
				max = toguess-1;
			}
			else if(toguess < x){//the guess was too low so Jimmy knows the minimum number x could be is more than toguess
				min = toguess+1;
			}


		}


		 //writing the output
		 cout << "Game #" << game << ": ";

		//printing differently if the number of guesses was one
		if(numberOfGuesses == 1) cout << numberOfGuesses << " guess" << endl;
		else cout << numberOfGuesses << " guesses" << endl;


	}
	
	return 0;
}
